src/Feeds.js

```diff
@@ -1,270 +1,425 @@
-    } catch (err) {
-      log.error(err)
-      throw new Error(_err.failedGettingActiveFeeds)
-    }
-
-    let res
-    try {
-      res = await Promise.all(feeds.map((user) => {
-        return this.slashtags.feed(user.user_id, {
-          announce: true
-        })
-      }))
-    } catch (err) {
-      console.log(err)
-      throw new Error(_err.failedBroadcast)
-    }
-    return {
-      feeds_started: res.length
-    }
-  }
+  async deleteFeed (args) {
+    if (!this.ready) throw new Err(_err.notReady)
+    if (!args?.feed_id) throw new Err(_err.feedIdMissing)
+    if (typeof args.feed_id !== 'string') throw new Err(_err.feedIdNotString)
 
-  async deleteUserFeed (args) {
-    if (typeof args.user_id !== 'string') throw new Err(_err.useridNotString)
-    const userId = args.user_id
+    const feedId = args.feed_id
     try {
-      const existingUser = await this.getFeedFromDb(userId)
-      if (!existingUser) {
-        log.info(`Deleting user that does not exist: ${userId}`)
+      const existingFeed = await this.getFeedFromDb(args)
+      if (!existingFeed) {
+        log.info(`Deleting feed that does not exist: ${feedId}`)
         return { deleted: true }
       }
-      await this.db.removeUser(userId)
-      await this.slashtags.destroy(userId)
+      // XXX: this needs to be atomic to prevent discrepancy between local DB and Hyperdrive
+      await this.db.removeFeed(feedId)
+      await this.slashtags.destroy(feedId)
     } catch (err) {
-      console.log(err)
-      throw new Error(_err.failedDeleteUser)
+      log.err(err)
+      if (err instanceof Err) throw err
+      throw new Err(_err.failedDeleteFeed)
     }
 
-    return {
-      deleted: true
-    }
+    return { deleted: true }
   }
 
   /**
-   * @desc get user feed by key
-   * @param {String} userId userId
-   * @returns UserFeed object
+   * @desc get feed by key
+   * @param {String} feedId feedId
+   * @returns FeedFeed object
    */
-  async getFeedKey (userId) {
-    let userFeed
+  async getFeedKey (args) {
+    if (!this.ready) throw new Err(_err.notReady)
+    if (!args.feed_id) throw new Err(_err.feedIdMissing)
+    if (typeof args.feed_id !== 'string') throw new Err(_err.feedIdNotString)
+    let feed
     try {
-      userFeed = await this.slashtags.feed(userId)
-      if (!userFeed.key) throw new Err(_err.userNoFeed)
+      feed = await this.slashtags.feed(args.feed_id)
+      if (!feed.key) throw new Err(_err.idNoFeed)
     } catch (err) {
       log.err(err)
       if (err instanceof Err) throw err
-      throw new Err(_err.userNoFeed)
+      throw new Err(_err.idNoFeed)
     }
     return {
-      key: userFeed.key.toString('hex'),
-      encryption_key: userFeed.encryptionKey.toString('hex')
+      // XXX should it be hex or base32
+      key: feed.key.toString('hex'),
+      encryption_key: feed.encryptionKey.toString('hex')
     }
   }
 
   async getFeed (args) {
+    if (!this.ready) throw new Err(_err.notReady)
```

Comment:
Fwiw, you don't need to store neither the key, nor the encryption key in a database, since `@synonymdev/feeds` always generate the same keys from whatever name you give it.

So I don't even think you need a user table because applications have their own, but I understand if you still want to control that.

Just know that you don't need to, this is more than enough `Feeds.feed(userID)`